org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLWorkspaceService.java

```diff
@@ -37,6 +46,25 @@ public XMLWorkspaceService(XMLLanguageServer xmlLanguageServer) {
 	public CompletableFuture<List<? extends SymbolInformation>> symbol(WorkspaceSymbolParams params) {
 		return null;
 	}
+	
+	
+	@Override
+	public CompletableFuture<Object> executeCommand(ExecuteCommandParams params) {
+		synchronized (commands) {
+			ExecuteCommandHandler handler = commands.get(params.getCommand());
+			return handler.handle(params);
+		}
+	}
+	
+	public void onCommand(String id, ExecuteCommandHandler commandHandler) {

```

Comment:
It seems onCommand is never used?

Reply:
It is used by my Spring XML extension. It registers/uregisters a command and need to provide a handler for the command. This is where the `onCommand` is used... I think though that you're in favour of me not accessing the server  from the extension directly. Would need to see how best to do it and then `onCommand` would most likely used in the XML server code.

Reply:
I would like to avoid exposing XMLWorkspaceService, XMLTextDocumentService and XMLLanguageServer. 

LemMinx is extensible via XMLExtensionRegistry (completion with ICompletionParticpant, etc), so I think we should do the same thing for Command. In other words, XMLExtensionRegistry should store the command and not the XMLLanguageServer.

To register a command, you create a LemMinx extension (see https://github.com/eclipse/lemminx/blob/master/docs/LemMinX-Extensions.md#creating-a-lemminx-extension for more information) like SpringPlugin, and you register your command in the start method and unregister it in the stop method.